clang/include/clang/Basic/Attr.td

```diff
@@ -1281,6 +1281,23 @@ def SYCLSpecialClass: InheritableAttr {
   let Documentation = [SYCLSpecialClassDocs];
 }
 
+def SYCLType: InheritableAttr {
+  let Spellings = [Clang<"sycl_type">];
```

Comment:
Clang spelling actually [implies] a few spellings: `__attribute__(attr)` and `[[clang::attr]]`, so at first glance there shouldn't be a conflict, but I would still vote for adding underscores, because I suppose that it is legal for user to make a macro named `sycl_type`

Reply:
I think it makes sense to define this in the `__sycl_detail__` namespace. I'll make the change. I was going over all the SYCL attributes and see we have a lot of internal attributes not in this namespace though. The inconsistency irks me. I guess a followup PR is required at some point to correct this.

 I also just discovered there is a documentation category `InternalOnly `which we should be using instead of `Undocumented`, for attributes we never plan to document. 

```
// This category is for attributes which have not yet been properly documented,
// but should be.
def DocCatUndocumented : DocumentationCategory<"Undocumented"> {
  let Content = [{
This section lists attributes which are recognized by Clang, but which are
currently missing documentation.
}];
}

// Attributes listed under the InternalOnly category do not generate any entry
// in the documentation.  This category should be used only when we _want_
// to not document the attribute, e.g. if the attribute has no spellings.
def DocCatInternalOnly : DocumentationCategory<"InternalOnly">;
```